great_expectations/expectations/metrics/column_map_metrics/column_values_unique.py

```diff
@@ -62,14 +63,7 @@ def _sqlalchemy_window(cls, column, _table, **kwargs):
                 source_table=_table,
                 column_name=column.name,
             )
-            if sqlalchemy.Engine and isinstance(sql_engine, sqlalchemy.Engine):
-                with sql_engine.connect() as connection:
-                    with connection.begin():
-                        connection.execute(sa.text(temp_table_stmt))
-            else:
-                # sql_engine is a connection
-                with sql_engine.begin():
-                    sql_engine.execute(sa.text(temp_table_stmt))
+            execution_engine.execute_query_in_transaction(sa.text(temp_table_stmt))
```

Comment:
This pattern can be extended to other metrics.

Reply:
Interesting that this an `sqlalchemy_batch_data` are our few `execute_query_in_transaction` usages. I'm just thinking to myself here: I wonder what makes theses ones different than the other metrics